### PR TITLE
Add llama finetuning test

### PR DESCRIPTION
**Pull request by A9isha**
# Description

Adding a Llama2-7B finetuning test for both stable and nightly MaxText version

# Tests

I have run all the MaxText tests in addition to this new test

**Instruction and/or command lines to reproduce your tests:** 
Upload to Airflow, and test.

**List links for your tests (use go/shortn-gen for any internal link):** ...
http://shortn/_1pw9SF5HJm

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run one-shot tests and provided workload links above if applicable. 
- [X] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: dags/multipod/maxtext_end_to_end.py
Comment (RissyRan) [inline dags/multipod/maxtext_end_to_end.py:39]: Thanks Anisha! Do you think it's better to just add this [script](https://github.com/google/maxtext/blob/main/end_to_end/llama_finetuning_test.sh) into [test_llama2](https://github.com/google/maxtext/blob/main/end_to_end/test_llama2.sh)? So no change is needed in this file.

cc @jonb377 as code owner

Comment (jonb377) [reply dags/multipod/maxtext_end_to_end.py:39]: I'd say it's better to keep these tests independent, so the `end_to_end` tests don't require any configuration.

Comment (A9isha) [reply dags/multipod/maxtext_end_to_end.py:39]: Could you please explain the thought a bit more @jonb377 ?

Comment (jonb377) [reply dags/multipod/maxtext_end_to_end.py:39]: Oh I just mean we should keep all the configuration in `maxtext` and run the test using `bash end_to_end/<test_script>` here in XLML. If we consolidate the two scripts, we'd need to pass in some configuration to the test script at runtime.

I like what you have here @A9isha

**Review by jonb377 — APPROVED** [review]
LGTM, thanks Anisha!

File: dags/multipod/maxtext_end_to_end.py
Comment (RissyRan) [reply dags/multipod/maxtext_end_to_end.py:39]: I am happy to approve this PR! I was thinking if it will be better if we consolidate two scripts, as `test_llama2` already has test for training, decode, and others. Wondering what's the purpose to separate fine-tuning to a different script? Or will it be better if we organize it in a way, i.e. separate training or decode from the test_llama2 as well?

I think it's @jonb377's decision.

**Review by RissyRan — APPROVED** [review]

File: dags/multipod/maxtext_end_to_end.py
Comment (A9isha) [reply dags/multipod/maxtext_end_to_end.py:39]: Let us discuss this offline. Let me start a thread

**Pull request merged**
